core/src/main/java/io/github/tcdl/adapters/AdapterFactory.java

```diff
@@ -19,7 +22,17 @@ public static AdapterFactory getInstance() {
 
     public Adapter createAdapter(BrokerAdapter brokerName, String topic, MsbConfigurations msbConfig) {
         if (brokerName == BrokerAdapter.AMQP) {
-            return new AmqpAdapter(topic, msbConfig);
+            try {
+                Class<?> clazz = Class.forName("io.github.tcdl.adapters.amqp.AmqpAdapter");
+                Constructor<?> constructor = clazz.getConstructor(String.class, MsbConfigurations.class);
```

Comment:
A lot of reflection here. Maybe adapters shouldn't be created via direct constructor invocation but via call to some factory? Then we could extract the interface of that factory and call it more conveniently. Also it will enforce the interface for adapter creation (in our case topic name and configuration instance).